### PR TITLE
fix: normalize scheme-less HTTP proxy settings

### LINKED ISSUE #8292: enhancement: auto-prepend http:// scheme to http_proxy config for httpx compatibility
### Background

AstrBot supports configuring an HTTP proxy via the `http_proxy` field in `cmd_config.json`. The value is then set as an environment variable in `core_lifecycle.py`:

```python
os.environ["http_proxy"] = proxy_config
```

This value is consumed by various libraries, notably **httpx** (which is used internally by the OpenAI SDK and many AstrBot provider modules).

### Problem

When a user configures the proxy **without a URL scheme** (e.g., `127.0.0.1:7890` instead of `http://127.0.0.1:7890`), the following issues arise:

- ✅ **curl** and the **requests** library handle this gracefully
- ❌ **httpx** throws `ValueError: Unknown scheme for proxy URL`, breaking all httpx-dependent features (Whisper STT, TTS, OpenAI API calls, etc.)

This creates an inconsistent user experience, as the same proxy value works with some tools but silently breaks others.

### Current state

The WebUI config schema does include a `hint` explaining the expected format (`http://127.0.0.1:7890`). However:

- Users who edit `cmd_config.json` directly never see this hint
- There is no code-level validation or defensive handling in `core_lifecycle.py`
- The error message (`ValueError: Unknown scheme for proxy URL`) is opaque to non-technical users

### Suggested fix

In `core_lifecycle.py`, before setting the proxy environment variables, add a simple scheme auto-completion:

```python
if "://" not in proxy_config:
    proxy_config = "http://" + proxy_config
```

This is a minimal, zero-side-effect change that makes the configuration more robust without breaking existing setups.

### Additional context

- AstrBot version: 4.25.1+
- OS: Linux
- No existing issues found on this topic
- The fix has been tested locally and works as expected

### PR DESCRIPTION
**Pull request by Last-emo-boy**
Fixes #8292

### Modifications / 改动点

- Trim the global HTTP proxy setting before exporting it.
- Default scheme-less proxy addresses to http://.
- Preserve explicit HTTP, HTTPS, and SOCKS proxy schemes and the existing empty-value cleanup.

- [x] This is NOT a breaking change. / 这不是一个破坏性变更。

### Screenshots or Test Results / 运行截图或测试结果

- uv run pytest -q tests/unit/test_core_lifecycle.py: 33 passed
- ruff format --check .: passed
- ruff check .: passed

---

### Checklist / 检查清单

- [x] This change is covered by the linked issue.
- [x] My changes have been well-tested, and verification results are provided above.
- [x] No new dependencies are introduced.
- [x] My changes do not introduce malicious code.

## Summary by Sourcery

Normalize HTTP proxy configuration before exporting environment variables.

Bug Fixes:
- Ensure scheme-less HTTP proxy settings are treated as http:// proxies instead of invalid values.
- Trim whitespace-only or empty HTTP proxy values to avoid exporting meaningless environment variables.

Enhancements:
- Align http_proxy and https_proxy environment variables to use a normalized, scheme-consistent proxy value.

Tests:
- Add parameterized tests verifying proxy normalization, including scheme-less, trimmed, explicit scheme, and empty/None configurations.

### CODE REVIEW

File: tests/unit/test_core_lifecycle.py [hunk 1]
```diff
@@ -77,6 +77,47 @@ def test_init_with_proxy(
...
+        ("configured_proxy", "expected_proxy"),
+        [
+            ("127.0.0.1:7890", "http://127.0.0.1:7890"),
+            ("  proxy.example.com:8080  ", "http://proxy.example.com:8080"),
+            (" https://proxy.example.com:8080 ", "https://proxy.example.com:8080"),
+            ("socks5://proxy.example.com:1080", "socks5://proxy.example.com:1080"),
+            ("   ", None),
+            (None, None),
+        ],
+    )
```
[the remark this answers is not in the document]
Comment (Last-emo-boy) [reply tests/unit/test_core_lifecycle.py:91]: Addressed in 54cc4ebb1: added coverage for an already-schemed HTTP proxy. The normalization remains inline and intentionally preserves any explicitly configured scheme; this is a single-use three-line path and follows the repository Inline-First rule.

File: astrbot/core/core_lifecycle.py [hunk 2]
```diff
@@ -65,7 +65,9 @@ def __init__(self, log_broker: LogBroker, db: BaseDatabase) -> None:
         self._default_chat_provider_warning_emitted = False
 
         # 设置代理
-        proxy_config = self.astrbot_config.get("http_proxy", "")
+        proxy_config = str(self.astrbot_config.get("http_proxy", "") or "").strip()
```
[the remark this answers is not in the document]
Comment (Last-emo-boy) [reply astrbot/core/core_lifecycle.py]: Addressed in 0ed1e2d33: removed unconditional str coercion. None still maps to an empty setting, strings are trimmed, and other types fail immediately on strip() instead of being converted into an invalid proxy URL. Because the config schema already declares http_proxy as a string, this preserves fail-fast behavior without adding a broader validation layer.
